release/train_tests/benchmark/train_benchmark.py

```diff
@@ -135,36 +163,51 @@ def validate_and_checkpoint(self):
 
     def validate(self) -> Dict[str, float]:
         if ray.train.get_context().get_world_rank() == 0:
-            print(
-                f"Validation starting @ epoch={self._train_epoch_idx}, batch={self._train_batch_idx}"
+            logger.info(
+                f"[Validation] Starting @ epoch={self._train_epoch_idx}, "
+                f"batch={self._train_batch_idx}"
             )
 
+        val_dataloader = self.factory.get_val_dataloader()
+
         self.model.eval()
 
         total_loss = torch.tensor(0.0).to(ray.train.torch.get_device())
         num_rows = 0
 
         with self._metrics["validation/iter_first_batch"].timer():
-            batch = self.get_next_batch(self.val_dataloader)
+            batch = self.get_next_batch(val_dataloader)
 
         while batch:
             input_batch, labels = batch
 
             with self._metrics["validation/step"].timer():
-                with torch.no_grad():
-                    out = self.model(input_batch)
-                    loss = self.loss_fn(out, labels)
-                    total_loss += loss
-                    num_rows += len(labels)
-                    self._metrics["validation/rows_processed"].add(len(labels))
+                if not self.benchmark_config.skip_validation_step:
+                    total_loss += self.validate_step(input_batch, labels)
```

Comment:
Inverse ordering of timer and condition? Or do you intentionally want to include this as well like you mentioned in the other comment.

```python
            if not self.benchmark_config.skip_validation_step:
                with self._metrics["validation/step"].timer():
                    total_loss += self.validate_step(input_batch, labels)
```

Reply:
I think it's easier to parse the output if this metric always exists and is just 0 rather than existing conditionally.